spec-main/api-autoupdater-darwin-spec.ts

```diff
@@ -0,0 +1,263 @@
+import { expect } from 'chai'
+import * as cp from 'child_process'
+import * as http from 'http'
+import * as express from 'express'
+import * as fs from 'fs-extra'
+import * as os from 'os'
+import * as path from 'path'
+import { AddressInfo } from 'net';
+
+const features = process.atomBinding('features')
+
+const fixturesPath = path.resolve(__dirname, '../spec/fixtures')
+
+// We can only test the auto updater on darwin non-component builds
+const describeFn = (process.platform === 'darwin' && !process.mas && !features.isComponentBuild() ? describe : describe.skip)
+
+describeFn('autoUpdater behavior', function () {
+  this.timeout(120000)
+
+  let identity = ''
+
+  beforeEach(function () {
+    const result = cp.spawnSync('bash', [path.resolve(__dirname, '../script/codesign/get-trusted-identity.sh')])
+    if (result.status !== 0 || result.stdout.toString().trim().length === 0)  {
+      if (isCI) {
+        throw new Error('No valid signing identity available to run autoUpdater specs')
+      }
+      this.skip()
+    } else {
+      identity = result.stdout.toString().trim()
+    }
+  })
+
+  it('should have a valid code signing identity', () => {
+    expect(identity).to.be.a('string').with.lengthOf.at.least(1)
+  })
+
+  const copyApp = async (newDir: string, fixture = 'initial') => {
+    const appBundlePath = path.resolve(process.execPath, '../../..')
+    const newPath = path.resolve(newDir, 'Electron.app')
+    cp.execSync(`cp -R "${appBundlePath}" "${path.dirname(newPath)}"`)
+    const appDir = path.resolve(newPath, 'Contents/Resources/app')
+    await fs.mkdirp(appDir)
+    await fs.copy(path.resolve(fixturesPath, 'auto-update', fixture), appDir)
+    const plistPath = path.resolve(newPath, 'Contents', 'Info.plist')
+    await fs.writeFile(
+      plistPath,
+      (await fs.readFile(plistPath, 'utf8')).replace('<key>BuildMachineOSBuild</key>', `<key>NSAppTransportSecurity</key>
+      <dict>
+          <key>NSAllowsArbitraryLoads</key>
+          <true/>
+          <key>NSExceptionDomains</key>
+          <dict>
+              <key>localhost</key>
+              <dict>
+                  <key>NSExceptionAllowsInsecureHTTPLoads</key>
+                  <true/>
+                  <key>NSIncludesSubdomains</key>
+                  <true/>
+              </dict>
+          </dict>
+      </dict><key>BuildMachineOSBuild</key>`)
+    )
+    return newPath
+  }
+
+  const spawn = async (cmd: string, args: string[], opts: any = {}) => {
+    let out = ''
+    const child = cp.spawn(cmd, args, opts)
+    child.stdout.on('data', (chunk: Buffer) => {
+      out += chunk.toString()
+    })
+    child.stderr.on('data', (chunk: Buffer) => {
+      out += chunk.toString()
+    })
+    return await new Promise<{ code: number, out: string }>((resolve) => {
```

Comment:
no need to `return await new Promise`, just drop the `async` keyword from the function definition and `return new Promise`. also might as well wrap the whole function in `new Promise`.